frontend/src/components/ModelBox/ModelConfiguration.tsx

```diff
@@ -45,21 +45,32 @@ function ModelConfiguration() {
 		},
 	]);
 
+	function setCustomChatModelByID(id: MODEL_CONFIG, value: number) {
+		setCustomChatModel((prev) => {
+			return prev.map((config) => {
+				if (config.id === id) {
+					return { ...config, value };
+				}
+				return config;
+			});
+		});
+	}
+
 	function updateConfigValue(id: MODEL_CONFIG, newValue: number) {
-		const selectedConfig = customChatModelConfigs.find(
+		const prevValue = customChatModelConfigs.find(
 			(config) => config.id === id
-		);
-		if (selectedConfig) {
-			const otherConfigs = customChatModelConfigs.filter(
-				(config) => config.id !== id
-			);
-			const updatedSelectedConfig = { ...selectedConfig, value: newValue };
-			const newCustomChatModelConfigs = [
-				updatedSelectedConfig,
-				...otherConfigs,
-			];
-			setCustomChatModel(newCustomChatModelConfigs);
-		}
+		)?.value;
+
+		if (!prevValue)
```

Comment:
Is it possible to have a falsy value for a config, e.g. empty string or zero? Probably not, but this can be future-proofed by explicitly checking for undefined. Or, by storing the config here instead of just its value, leaving it as a falsy check, and using config.value below when resetting the value. Up to you.

Reply:
Ah! Good shout. This component only deals with slider, and yes it is possible, probable even, that we'd want to set one to 0.